### DISCUSSION TITLE
Conditionally check for TOC

### DISCUSSION THREAD

**Discussion by jeremyzilar** [allejo/jekyll-toc#34] [Q&A] [answered]
**Public Repo** https://github.com/brooklynrail/brooklynrail-handbook/

**Description**

I would like to be able to wrap the TOC is some markup.
```
<div class="toc">
  <h2>On this page</h2>
  ...toc-goes-here...
</div>
```

but I am having a hard time figuring out how to conditionally check to see if there is a TOC, or if there are headings on the page.

The problem is that the TOC `<div>` and `<h2>` still appear on the page when no headings used.

Do you mind pointing out where I would conditionally check for the TOC in the file?

**Comment by allejo** [comment, marked as answer]
Untested code but you'd want to use the following logic:

1. Capture the rendered TOC
2. Strip any whitespace around the TOC
3. Check to see if a TOC was rendered. If there was, render the surrounding `<div>` and use the TOC that was captured

```twig
{% capture page_toc %}
  {% include toc.html html=content %}
{% endcapture %}
{% assign page_toc = page_toc | strip %}

{% if page_toc != "" %}
  <div class="toc">
    <h2>On this page</h2>
    {{ page_toc }}
  </div>
{% endif %}
```

**Comment by jeremyzilar** [comment]
Thanks. I’ll give that a try
